### PR TITLE
Fix query profiler test basic with concurrent execution

### LINKED ISSUE #10941: [BUG] org.opensearch.search.profile.query.QueryProfilerTests.testBasic {p0=5} is flaky
**Describe the bug**
`org.opensearch.search.profile.query.QueryProfilerTests.testBasic {p0=5}` test is flaky.

Below assertion failure can be seen -
```
java.lang.AssertionError: 
Expected: a value greater than <0L>
     but: <0L> was equal to <0L>
```

**To Reproduce**
https://build.ci.opensearch.org/job/gradle-check/29044/testReport/junit/org.opensearch.search.profile.query/QueryProfilerTests/testBasic__p0_5_/

**Expected behavior**
Test must always pass.

### PR DESCRIPTION
**Pull request by ticheng-aws**
### Description
This PR attempts to fix the `QueryProfilerTests.testBasic` with concurrent execution. The issue happened where a slice has `0` invocations for a breakdown type from its leading leaves. For example, a slice with three leaves: leaf A with a next_doc count of  `0`, leaf B with a next_doc count of  `0`, and leaf C with a next_doc count of  `4`. We should skip the slice start/end time computation for leaf A and leaf B since `0` does not represent an actual timing. Only when we've checked all leaves in a slice and still find no invocations, then we should set the slice start/end time to the default `0`.

### Related Issues
Resolves #10941

### Check List
- [ ] ~New functionality includes testing.~
  - [X] All tests pass
- [ ] ~New functionality has been documented.~
  - [ ] ~New functionality has javadoc added~
- [X] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [X] Commits are signed per the DCO using --signoff
- [ ] ~Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))~
- [ ] ~Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java [hunk 1]
```diff
@@ -202,6 +202,11 @@ Map<Collector, Map<String, Long>> buildSliceLevelBreakdown() {
                         // doesn't represent an actual timing; rather, it indicates no invocations.
                         continue;
                     }
+                    if (sliceLeafTimingTypeStartTime == 0L && currentSliceBreakdown.get(timingTypeCountKey) == 0L) {
```
Comment (sohami) [inline server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java:205]: This check and the one above on line 197 is essentially ignoring the leaf with 0 invocation for a timing type. We can merge these and instead add a check if the timingTypeCount is 0 then ignore that leaf timingType values.

File: server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java [hunk 2]
```diff
@@ -216,6 +221,12 @@ Map<Collector, Map<String, Long>> buildSliceLevelBreakdown() {
                         (key, value) -> (value == null) ? sliceLeafTimingTypeEndTime : Math.max(value, sliceLeafTimingTypeEndTime)
                     );
                 }
+                // Only when we've checked all leaves in a slice and still find no invocations, then we should set the slice start/end time
+                // to the default 0L.
```
Comment (sohami) [inline server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java:225]: Lets add to the comment that `buildQueryBreakdownMap` expects `timingTypeSliceStartTimeKey` and `timingTypeSliceEndTimeKey` in the slice level breakdowns.

File: server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java (hunk 1 above)
Comment (ticheng-aws) [reply server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java]: Hi @sohami, thanks for the review. It's been addressed now.

File: server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java (hunk 2 above)
Comment (ticheng-aws) [reply server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java]: It has been added now.

**Review by sohami — APPROVED** [review]
lgtm

**Pull request merged**